hack/update-codegen.sh

```diff
@@ -1,38 +1,134 @@
+                ;;
+            "--verify" )
+                VERIFY=1
+                ;;
+            * )
+                echo "error: invalid command line option '$1'" >&2
+                exit 1
+        esac
+        shift
+    done
+}
+
+function print_usage() {
+    echo "usage:"
+    echo ""
+    echo "   $(basename $BASH_SOURCE) [OPTIONS]"
+    echo ""
+    echo "When run without any options, all existing generated code will be regenerated."
+    echo ""
+    echo "OPTIONS"
+    echo ""
+    echo "   -h, --help"
+    echo "      Print usage help and exit. No other operations will be performed."
+    echo ""
+    echo "   --verify"
+    echo "      Verifies that the generated code is up-to-date. The existing generated"
+    echo "      code will not be touched."
+    echo ""
+}
+
+function is_verify_mode() {
+    [[ -n ${VERIFY:-} ]]
+}
+
+
+#
+# main
+#
+
+source "$HERE/.setpaths"
+
+read_args "$@"
+
+# Check and prepare build enviroment
+GOPATH=`go env GOPATH`
 if [[ -z $GOPATH ]]; then
-    echo "error: GOPATH not set"
-    exit 1
+    die "GOPATH not set"
+fi
+GOPATH_1=${GOPATH%%:*}  # the first entry of the GOPATH
+
+# prepare code generator
+"$HERE/bootstrap-codegen.sh" || die "failed to bootstrap code generator"
+[[ -f $CODEGEN_PKG/generate-groups.sh ]] \
+    || die "\$CODEGEN_PKG ('$CODEGEN_PKG'): file 'generate-groups.sh' does not exist"
+[[ -x $CODEGEN_PKG/generate-groups.sh ]] \
+    || die "\$CODEGEN_PKG ('$CODEGEN_PKG'): file 'generate-groups.sh' is not executable"
+
+# prepare mockgen
+MOCKGEN_EXE="$GOPATH_1/bin/mockgen"
+if [[ ! -x $MOCKGEN_EXE ]]; then
+    echo "Installing mockgen"
+    ( cd "$GOPATH_1" && go get github.com/golang/mock/mockgen ) || die "Installation of mockgen failed"
 fi
+[[ -f $MOCKGEN_EXE ]] || die "'$MOCKGEN_EXE' does not exist"
+[[ -x $MOCKGEN_EXE ]] || die "'$MOCKGEN_EXE' is not executable"
 
-PROJECT_ROOT=$(cd "$(dirname "$BASH_SOURCE")/.."; pwd)
 GEN_DIR="$PROJECT_ROOT/gen"
-GOPATH_1=${GOPATH%%:*}  # the first entry of the GOPATH
+
+if is_verify_mode; then
+    MOCK_ROOT=${GEN_DIR}
+    ACTION="Verify"
+else    
+    MOCK_ROOT=${PROJECT_ROOT}
+    ACTION="Generate"
+fi
 
 echo
 echo "PROJECT_ROOT: $PROJECT_ROOT"
 echo "GEN_DIR:      $GEN_DIR"
+echo "MOCK_ROOT:    $MOCK_ROOT"
 echo "CODEGEN_PKG:  $CODEGEN_PKG"
+echo "GOPATH:       $GOPATH_1"
+echo "VERIFY:       $(if is_verify_mode; then echo "true"; else echo "false"; fi)"
 
 echo
 echo "## Cleanup old generated stuff ####################"
-set -x
-rm -rf \
-    "${PROJECT_ROOT}/pkg/client" \
-    "${PROJECT_ROOT}/pkg/tektonclient" \
-    "${PROJECT_ROOT}/pkg/apis/steward/v1alpha1/zz_generated.deepcopy.go" \
-    "${PROJECT_ROOT}/pkg/k8s/mocks/mocks.go" \
-    "${GEN_DIR}/github.com" \
-    "${GOPATH_1}/bin/"{client-gen,deepcopy-gen,defaulter-gen,informer-gen,lister-gen}
-set +x
+if is_verify_mode; then
+    set -x
+    rm -rf \
+        "${GEN_DIR}" \
+        "${GOPATH_1}/bin/"{client-gen,deepcopy-gen,defaulter-gen,informer-gen,lister-gen} \
```

Comment:
This line is the same in both if branches. Use a separate `rm` call outside the `if` for common parts.

Reply:
see https://github.com/SAP/stewardci-core/pull/19/commits/0b66a2e2d217d90c2e4b8861511e1066d9b3c376